src/status.rs

```diff
@@ -129,7 +129,10 @@ impl Status {
         header_map.insert(GRPC_STATUS_HEADER_CODE, self.code.to_header_value());
 
         if !self.message.is_empty() {
-            header_map.insert(GRPC_STATUS_MESSAGE_HEADER, HeaderValue::from_str(&self.message)
+            header_map.insert(GRPC_STATUS_MESSAGE_HEADER, HeaderValue::from_str(
+                    Box::leak(utf8_percent_encode(&self.message(), DEFAULT_ENCODE_SET)
```

Comment:
Why use `Box::leak` here? That sounds like it will.. leak memory, no? Isn't it easier to do

    utf8_percent_encode(...).to_string().parse().unwrap()

instead of the `HeaderValue::from_str(...)` call (maybe also handle the potential failure there)

Reply:
- To save on allocations, the owned string can be converted into a `Bytes` and then [`HeaderValue::from_shared`](https://docs.rs/http/0.1.16/http/header/struct.HeaderValue.html#method.from_shared) can be called.
- Looking at other grpc implementations, it looks like we want to define an encode set that encodes anything outside of the visible ASCII range.

Reply:
Thanks for reviewing! About the encoding set for Status message, I browsed the grpc-java and grpc-go. Looks like they directly use percent encoding - the byte is converted into hexadecimal notation with a '%' prepended without custom encoding set.
But for some Unicode char like emoji, `percent_encoding::decode_utf8()` sometimes can't give correct parsing. E.g.
```
let s1 = Status::new(Code::Unknown, "Unicode non-BMP 🌶️").to_header_map();
let s2 = Status::from_header_map(&s2);
```
will give s2 as `Status { code: Unknown, message: "Unicode non-BMP 🌶\u{fe0f}" }` which `\u{fe0f}` should not appear.

Reply:
Also for adding unit tests for `Status`, is that necessary to implement `PartialEq` for `Status` in order to use assert?

Reply:
No, there shouldn't be need to add `PartialEq for Status` (in fact, that's probably not a great idea in general).

Should be good enough to just use two asserts, `assert_eq!(status.code(), expected_code)` and `assert_eq!(status.message(), expected_msg)`.